quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/themes/QuarkusFolderThemeProviderFactory.java

```diff
@@ -0,0 +1,50 @@
+package org.keycloak.quarkus.runtime.themes;
+
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.quarkus.runtime.Environment;
+import org.keycloak.theme.FolderThemeProvider;
+import org.keycloak.theme.ThemeProvider;
+import org.keycloak.theme.ThemeProviderFactory;
+
+import java.io.File;
+
+public class QuarkusFolderThemeProviderFactory implements ThemeProviderFactory {
+
+    private FolderThemeProvider themeProvider;
+
+    @Override
+    public ThemeProvider create(KeycloakSession sessions) {
+        return themeProvider;
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+        String d = config.get("dir");
+        File rootDir = null;
+        if (d != null) {
+            rootDir = new File(d);
+        } else {
+            //assume standard path as fallback
+            rootDir = new File(Environment.getDefaultThemeRootDir());
+        }
+        themeProvider = new FolderThemeProvider(rootDir);
```

Comment:
I know this is something we don't do in the original implementation. But in order to avoid mistakes we should probably check if the dir exists and if not throw some exception?

Reply:
or perhaps create it via mkdir()?

Reply:
I don't think we should create the dir but expect it exists. That is always true by default but if users provider their own, they should make sure the dir is there ...

Reply:
I don't think we should throw an exception if the dir doesn't exist, as it's not strictly required unless there's custom themes deployed there.

Reply:
pushed a new version, only throws an exception when themedir is set up in config but not accessible, OR when default theme dir is not accessible. 

For the sake of completeness, my work in progress solution was a bit more fallback-y (and thus, a little more verbose) -> 

```
private File createOrRetrieveThemeRootDir(String rootDirFromConfig) {
        File themeRootDir = null;

        try {
            if (rootDirFromConfig != null) {
                themeRootDir = new File(rootDirFromConfig);

                //rootdir specified in config does not exist. use default fallback.
                if (!themeRootDir.exists()) {
                    boolean created = new File(Environment.getDefaultThemeRootDir()).mkdir();

                    Logger.getLogger(QuarkusFolderThemeProviderFactory.class)
                            .warnf("Could not get theme root directory '%s' specified in config. " +
                                    (created ? "Created new default theme root directory at %s" : "Using default directory at %s instead."), rootDirFromConfig, Environment.DEFAULT_THEMES_PATH);
                }
            } else {
                //assume standard path as fallback
                themeRootDir = new File(Environment.getDefaultThemeRootDir());
                
                //someone deleted the /themes folder. re-create it.
                if (!themeRootDir.exists()) {
                    themeRootDir = new File(Environment.getDefaultThemeRootDir());
                    themeRootDir.mkdir();
                }
            }
            //only if that fails, throw runtimeexception.
        } catch (SecurityException e) {
            throw new RuntimeException("It was not possible to create theme root directory either from config or from default location.", e);
        }
        return themeRootDir;
    }
```
and would've tried to create the default dir if it does not exist. 

@pedroigor @stianst  If you want, I can remove the exceptions and return null and a log warning instead.

Reply:
Added new changes: if config value exists use config, else use default. if file doesnt exist, return null (as we talked about earlier). No Warn/exception included.

Reply:
@stianst Ideally we should fail fast and give useful messages when the server is building/starting. The idea was to avoid unexpected behavior at runtime due to a bad configuration.

It should be fine to keep it behaving as today but perhaps we should be discussing more how to improve this and some other providers.